src/App.tsx

```diff
@@ -17,18 +49,39 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filter={filter}
+                setFilter={setFilter}
+                query={query}
+                setQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length === 0 ? (
```

Comment:
If we have a todo length of 0, then the loader will be constantly displayed, you need to create a state for the loader
```suggestion
              {isLoading ? (
```